src/null-safety/index.md

```diff
@@ -1,41 +1,314 @@
 ---
 title: Null safety
-description: Watch this space for news about null safety
-toc: false
+description: Information about Dart's upcoming null safety feature
 ---
 
-Watch this page for updates about null safety.
-In the meantime, check out these resources:
+Sound null safety is coming to the Dart language!
+When you opt into null safety,
+types in your code are non-nullable by default, meaning that
+values can’t be null _unless you say they can be._
+With null safety, your **runtime** null-dereference errors
+turn into **edit-time** analysis errors.
+
+With null safety,
+the Dart analyzer enforces good practices.
+For example, it makes sure you check for null before
+reading a nullable variable.
+And because Dart null safety is sound,
+Dart compilers and runtimes can optimize away internal null checks,
+so apps can be faster and smaller.
+
+{{ site.alert.important }}
+  Because null safety is still in tech preview,
+  **don't use null safety in production code.**
+  But please do test the feature and [give us feedback.][]
+{{ site.alert.end }}
+
+New operators and keywords related to null safety
+include `?`, `!`, and `late`.
+If you've used Kotlin, TypeScript, or C#,
+the syntax for null safety might look familiar.
+That's by design: the Dart language aims to be unsurprising.
+
+You can practice using null safety in the web app
+[DartPad with Null Safety,][nullsafety.dartpad.dev]
+shown in the following screenshot.
+Or try null safety in your normal development environment,
+using the instructions and configuration files in the
+[null safety sample.][calculate_lix]
+
+![Screenshot of DartPad null safety snippet with analysis errors](/null-safety/dartpad-snippet.png)
+{% comment %}
+[TODO: update that screenshot]
+{% endcomment %}
+
+
+## Creating variables
+
+When creating a variable,
+you can use `?` and `late`
+to inform Dart of the variable's nullability.
+
+Here’s an example of declaring a **non-nullable integer variable**
+(assuming you’ve opted into null safety):
+
+```dart
+int aRealInt; // Can't be used until it's initialized to an integer.
```

Comment:
I'm in doubt about this one. Given it's the very first example of a non-nullable variable on the page, should we not show what I think is the more regular case: initializing it with a value?

```dart
int aRealInt = 42; // Can't be assigned null.
```

Reply:
If it's assigned a value, wouldn't you normally use `var` instead?

Reply:
Yup. The blog post has these examples:

```dart
// In null-safe Dart, none of these can ever be null.
var i = 42;
final b = Foo();
String m = '';
```